pallets/task/src/lib.rs

```diff
@@ -471,26 +501,98 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Function to revive an expired task. [origin, task_id, new_deadline]
+		/// Something the user does to allow editing of the task as well as keep the task in storage.
+		#[pallet::weight(<T as Config>::WeightInfo::revive_task(0,0))]
+		pub fn revive_expired_task(origin: OriginFor<T>, task_id: T::Hash, new_deadline: u64) -> DispatchResult {
```

Comment:
For now its ok to only change the deadline. Later we can change other items as well.